src/qt/android/src/org/bitcoincore/qt/BitcoinQtActivity.java

```diff
@@ -0,0 +1,29 @@
+package org.bitcoincore.qt;
+
+import android.os.Bundle;
+import android.system.ErrnoException;
+import android.system.Os;
+
+import org.qtproject.qt5.android.bindings.QtActivity;
+
+import java.io.File;
+
+public class BitcoinQtActivity extends QtActivity
+{
+    @Override
+    public void onCreate(Bundle savedInstanceState)
+    {
+        final File bitcoinDir = new File(getFilesDir().getAbsolutePath() + "/.bitcoin");
+        if (!bitcoinDir.exists()) {
+            bitcoinDir.mkdir();
+        }
```

Comment:
This code prevents the Intro dialog from showing at the first startup. Is it intended? If yes, why?

Reply:
TBH I don't remember why I did this. Did you try without it? Maybe the dialog was showing incorrectly?

I didn't at the time but I realise now that this makes using external storage impossible. Possibly related to https://github.com/bitcoin-core/gui/pull/496